leaf-ui/js/displayAnalysis.js

```diff
@@ -239,4 +252,147 @@ function updateHistory(currentAnalysis){
+/**
+ * Adds a new analysisConfig to the sidebar
+ */
+function addAnalysisConfig(){
+    $(".log-elements").css("background-color", "");
+    $(".result-elements").css("background-color", "");
+    // Update current config with current analysisRequest and set the udpated config in map
+    currAnalysisConfig.updateAnalysis(analysisRequest);
+    analysisMap.set(currAnalysisConfig.id, currAnalysisConfig);
+
+    // Figure out number of new config, name and create it, and then add it to the map
+    var numElements = analysisMap.size;
+    var id = "Configuration" + (numElements+1).toString()
+    
+    // default Analysis Request needed for now for user assignments list
+    // TODO: Look into perserving base UAL throughout analysisRequests
+    var newRequest = new AnalysisRequest();
+    newRequest.userAssignmentsList = defaultUAL;
+
+    var newConfig = new AnalysisConfiguration(id, newRequest);
+    analysisMap.set(id, newConfig);
+
+    // Update current config to be the new config, and update analysisRequest to match new config
+    currAnalysisConfig = newConfig;
+    analysisRequest = currAnalysisConfig.getAnalysisRequest();
+
+    // Reset analysis sidebar to default
+    updateAnalysisBar();
+
+    var buttonLabel = currAnalysisConfig.id + "-button";
+    var label = currAnalysisConfig.id + "-dropdown";
+    $("#analysis-sidebar").append(
+        '<button class="log-elements" id="'+currAnalysisConfig.id+'" style="padding: 12px; font-size: 16px; border: none; outline: none; background-color:#A9A9A9">' + currAnalysisConfig.id + '</button><div style="position:absolute; display:inline-block"><button class="dropdown" id= "'+buttonLabel+'" style="padding: 12px; font-size: 16px; height: 42px; border: none; outline: none;"><i class="fa fa-caret-down fa-2x" style="cursor:pointer;"></i></button>'
+        + '</div><div class = "dropdown-container" id="'+label+'"></div>'
+      );
+
+    console.log(analysisRequest.userAssignmentsList);
+}
+
+/**
+ * Adds result to UI menu
+ */
+function updateResults(){
+    $(".result-elements").css("background-color", "");
+    var label = currAnalysisConfig.id + "-dropdown";
+    var resultCount = analysisMap.get(currAnalysisConfig.id).analysisResults.length;
+    var id = "Result " + resultCount;
+    document.getElementById(label).insertAdjacentHTML("beforeend","<a class='result-elements' style='background-color:#A9A9A9''>" + id + "</a>");
+}
+
+function updateAnalysisBar(){
+    $('#conflict-level').val(analysisRequest.conflictLevel);
+    $('#num-rel-time').val(analysisRequest.numRelTime);
+    document.getElementById('conflict-level').value = analysisRequest.conflictLevel;
+}
+
+/**
+ * Switches between analysis configurations
+ */
+$('#analysis-sidebar').on("click", ".log-elements", function(e){
+    //Save and/or update past analysis config in map
+    currAnalysisConfig.updateAnalysis(analysisRequest)
+    analysisMap.set(currAnalysisConfig.id, currAnalysisConfig);
+    currentConfigHtml = e.target;
```

Comment:
this isn't a critique I just want you to maybe explain to me how this works because html really confuses me lol - I saw you initialized this global var up above, what exactly is it for? is it like the button location or something?

Reply:
Yeah it basically tracks like what html element is being clicked currently based on the event e. But I actually think its not necessary, I ended up streamlining the html stuff a tiny bit so can delete. Thanks for pointing out!